packages/contracts/src.ts/index.ts

```diff
@@ -651,11 +652,11 @@ export class BaseContract {
     // Wrapped functions to call emit and allow deregistration from the provider
     _wrappedEmits: { [ eventTag: string ]: (...args: Array<any>) => void };
 
-    constructor(addressOrName: string, contractInterface: ContractInterface, signerOrProvider?: Signer | Provider) {
```

Comment:
I think that it would be better to rename `addressorName` to `accountLike` and make it optional and use `AccountLike` instead of `string`

Reply:
This way we must permute the arguments order - required parameters are always first. The new signature would be 
`abi, address?, signer?`

Reply:
Lets keep the interface but use default value for address